server/serve.go

```diff
@@ -23,11 +24,38 @@ func getQuestionName(z *zones.Zone, fqdn string) string {
 	return strings.ToLower(strings.Join(ql, "."))
 }
 
+func getIPFromDomain(dashedip string) (net.IP, error) {
+	i := strings.Split(dashedip, ".")[0]
+	p := strings.ReplaceAll(i, "-", ".")
+	ip := net.ParseIP(p)
+	if ip == nil {
+		return nil, errors.New("Invalid IP from domain address")
+	}
+	return ip, nil
+}
+
 func (srv *Server) serve(w dns.ResponseWriter, req *dns.Msg, z *zones.Zone) {
 
 	qnamefqdn := req.Question[0].Name
 	qtype := req.Question[0].Qtype
 
+	if qtype == dns.TypeA && z.ParseIP == true {
+		m := new(dns.Msg)
+		m.SetReply(req)
+
+		ip, err := getIPFromDomain(qnamefqdn)
+		if err != nil {
+			applog.Printf("Error writing packet: %q, %s", err, m)
+			dns.HandleFailed(w, req)
+		}
+
+		h := dns.RR_Header{Name: qnamefqdn, Rrtype: 1, Class: 1, Ttl: 120, Rdlength: 0}
```

Comment:
We should use a very long ttl here because the result will never change

Reply:
Fixed